src/brainbrowser/volume-viewer/volume-loaders/hdf5.js

```diff
@@ -0,0 +1,2355 @@
+      r.dims = [];              // dimension sizes.
+      return r;
+    }
+
+    /* Turns out that alignment of the messages in at least the
+     * version 1 object header is actually relative to the start
+     * of the header. So we update the start position of the
+     * header here, so we can refer to it when calculating the
+     * alignment in checkAlignment().
+     */
+    function startAlignment() {
+      start_offset = dv_offset;
+    }
+
+    function checkAlignment() {
+      var tmp = dv_offset - start_offset;
+      if ((tmp % align) !== 0) {
+        var n = align - (tmp % align);
+        dv_offset += n;
+        if (debug) {
+          console.log('skipping ' + n + ' bytes at ' + tmp + ' for alignmnent');
+        }
+      }
+    }
+
+    /* helper functions to manipulate the current DataView offset.
+     */
+    function skip(n_bytes) {
+      dv_offset += n_bytes;
+    }
+
+    function seek(new_offset) {
+      dv_offset = new_offset;
+    }
+
+    function tell() {
+      return dv_offset;
+    }
+
+    /* helper functions for access to our DataView. */
+    function getU8() {
+      var v = dv.getUint8(dv_offset);
+      dv_offset += 1;
+      return v;
+    }
+    function getU16() {
+      var v = dv.getUint16(dv_offset, little_endian);
+      dv_offset += 2;
+      return v;
+    }
+    function getU32() {
+      var v = dv.getUint32(dv_offset, little_endian);
+      dv_offset += 4;
+      return v;
+    }
+    function getU64() {
+      var v = dv.getUint64(dv_offset, little_endian);
+      dv_offset += 8;
+      return v;
+    }
+    function getF32() {
+      var v = dv.getFloat32(dv_offset, little_endian);
+      dv_offset += 4;
+      return v;
+    }
+    function getF64() {
+      var v = dv.getFloat64(dv_offset, little_endian);
+      dv_offset += 8;
+      return v;
+    }
+    function getOffset(offsz) {
+      var v = 0;
+      offsz = offsz || superblk.offsz;
+      if (offsz === 4) {
+        v = dv.getUint32(dv_offset, little_endian);
+      } else if (offsz === 8) {
+        v = dv.getUint64(dv_offset, little_endian);
+      } else {
+        throw new Error('Unsupported value for offset size ' + offsz);
+      }
+      dv_offset += offsz;
+      return v;
+    }
+    function getLength() {
+      var v = dv.getUint64(dv_offset, little_endian);
+      dv_offset += superblk.lensz;
+      return v;
+    }
+    function getString(length) {
+      var r = "";
+      var i;
+      var c;
+      for (i = 0; i < length; i += 1) {
+        c = getU8();
+        if (c === 0) {
+          dv_offset += (length - i - 1);
+          break;
+        }
+        r += String.fromCharCode(c);
+      }
+      return r;
+    }
+
+    function getArray(typ, n_bytes, new_off) {
+      var value;
+      var n_values;
+      var new_abuf;
+      var i;
+      var spp = dv_offset;
+      if (new_off) {
+        dv_offset = new_off;
+      }
+      switch (typ) {
+      case type_enum.INT8:
```

Comment:
Nitpick. Can you properly indent the case statement ? Or you indent like that due to jshint ?